include/gfx/pal_sorting.hpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * This file is part of RGBDS.
+ *
+ * Copyright (c) 2022, Eldred Habert and RGBDS contributors.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#ifndef RGBDS_GFX_PAL_SORTING_HPP
+#define RGBDS_GFX_PAL_SORTING_HPP
+
+#include <vector>
+
+class Palette;
+
+namespace sorting {
```

Comment:
Not sure how I feel about per-file namespaces for these fairly simple programs, vs just using function names like `sort_indexed`, `sort_grayscale`, `sort_rgb` instead.

Reply:
It helps thematically grouping code.